test/jdk/java/io/IO/IO.java

```diff
@@ -0,0 +1,153 @@
+import org.junit.jupiter.api.condition.OS;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+/*
+ * @test
+ * @bug 8305457
+ * @summary java.io.IO tests
+ * @library /test/lib
+ * @run junit IO
+ */
+public class IO {
+
+    @ParameterizedTest
+    @ValueSource(strings = {"println", "print"})
+    public void printTest(String mode) throws Exception {
+        var file = Path.of(System.getProperty("test.src", "."), "Output.java")
+                .toAbsolutePath().toString();
+        var pb = ProcessTools.createTestJavaProcessBuilder("--enable-preview", file, mode);
+        OutputAnalyzer output = ProcessTools.executeProcess(pb);
+        assertEquals(0, output.getExitValue());
+        output.reportDiagnosticSummary();
+        String out = output.getOutput();
+        // The first half of the output is produced by Console, the second
+        // half is produced by IO: those halves must match.
+        // Executing Console and IO in the same VM (as opposed to
+        // consecutive VM runs, which are cleaner) to be able to compare string
+        // representation of objects.
+        assertFalse(out.isBlank());
+        assertEquals(out.substring(0, out.length() / 2),
+                out.substring(out.length() / 2));
+    }
+
+    /*
+     * printTest tests a _default_ console, which is normally
+     * jdk.internal.org.jline.JdkConsoleProviderImpl, whereas this test tests
+     * jdk.internal.io.JdkConsoleImpl. Those console implementations operate
+     * in different conditions and, thus, are tested separately.
+     *
+     * To test jdk.internal.io.JdkConsoleImpl one needs to ensure that both
+     * conditions are met:
+     *
+     *   - a non-existent console provider is requested
+     *   - isatty is true
+     *
+     * To achieve isatty, the test currently uses the EXPECT(1) Unix command,
+     * which does not work for Windows. Later, a library like pty4j or JPty
+     * might be used instead EXPECT, to cover both Unix and Windows.
+     */
+    @ParameterizedTest
+    @EnabledOnOs({OS.LINUX, OS.MAC})
+    @ValueSource(strings = {"println", "print"})
+    public void outputTestInteractive(String mode) throws Exception {
+        var expect = Paths.get("/usr/bin/expect"); // os-specific path
+        if (!Files.exists(expect) || !Files.isExecutable(expect)) {
+            throw new SkippedException("'" + expect + "' not found");
+        }
+        var testSrc = System.getProperty("test.src", ".");
+        OutputAnalyzer output = ProcessTools.executeProcess(
+                "expect",
+                Path.of(testSrc, "output.exp").toAbsolutePath().toString(),
+                System.getProperty("test.jdk") + "/bin/java",
+                "--enable-preview",
+                "-Djdk.console=gibberish",
```

Comment:
The test comment suggests this test is testing the default console implementation, but the invocation specifies `-Djdk.console=gibberish` which defaults to java.base. Is this what you intended?

Reply:
That comment says that this test tests jdk.internal.io.JdkConsoleImpl, which belongs to java.base. But, if you read it the way you described, I should definitely rephrase it.

Reply:
Sorry, I read it wrong. Your comment is clear so no need for rewording

Reply:
> Sorry, I read it wrong. Your comment is clear so no need for rewording

Still, I think that your misreading is a symptom of a problem with my wording. Let me try to rephrase it; let's see if you like it more.